custom_components/atrea/climate.py

```diff
@@ -349,43 +389,18 @@ def set_hvac_mode(self, hvac_mode):
         self.manualUpdate()
 
     def set_preset_mode(self, preset):
-        mode = False
-        program = False
-        if preset == "Off":
-            self.turn_off()
-        elif preset == "Automat":
-            mode = 1
-            program = 0
-        elif preset == "Ventilation":
-            mode = 2
-            program = 0
-        elif preset == "Circulation and Ventilation":
-            mode = 3
-            program = 0
-        elif preset == "Circulation":
-            mode = 4
-            program = 0
-        elif preset == "Night precooling":
-            mode = 5
-            program = 0
-        elif preset == "Disbalance":
-            mode = 6
-            program = 0
-        elif preset == "Overpressure":
-            mode = 7
-            program = 0
-        else:
-            _LOGGER.warn("Chosen preset=%s(%s) is incorrect preset.", str(preset),
-                         str(mode))
-
-        if(((self.air_handling_control == 'Manual' and program != 0)
-                or (self.air_handling_control == 'Schedule' and program != 1)
-                or (self.air_handling_control == 'Temporary' and program != 2))
-                and program != None):
-            self.atrea.setProgram(program)
+        mode = None
+        try:
+            mode = AtreaMode(ALL_PRESET_LIST.index(preset))
+        except ValueError:
+            _LOGGER.warn("Chosen preset=%s is incorrect preset.", str(preset))
+            return
 
-        if((mode != False and self._current_preset != mode)
-           or self.air_handling_control == 'Schedule'):
+        if mode == AtreaMode.OFF:
+            self.turn_off()
+        if self.atrea.getProgram() != AtreaProgram.TEMPORARY:
+            self.atrea.setProgram(AtreaProgram.TEMPORARY)
```

Comment:
I like the refactor, however, it should default to MANUAL mode instead of Temporary. Thats how it was working before.

Reply:
That's somewhat related to the point above. My use case is that I leave my unit in AUTOMATIC and only sometimes switch it to TEMPORARY and override fan speed and preset. That way the unit performs ventilation and then reverts back to AUTOMATIC after a few minutes. If I were to set it to MANUAL, it would keep those overrides indefinitely.
So I guess introducing the separate `set_program()` method (and avoiding setting a program from `set_fan_mode()`, `set_preset_mode()` and `set_hvac_mode()`) would be the most flexible approach.

Reply:
Would the above solution work here too?

Reply:
Yes, done.